### PR TITLE
Remove some usage of unstable features

### PR DESCRIPTION
**Pull request by SimonSapin**
`Box::new(x)` seems to compile to exactly the same assembly as `box x` in release mode: 
https://play.rust-lang.org/?gist=55ec5dd02c22fa799375d6e2e1a6e970&version=nightly

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18839)

### CODE REVIEW

**Comment by nox** [comment]
IIRC removing `box` caused perf regressions on Dromaeo in the past.

File: components/net/image_cache.rs [hunk 1]
```diff
@@ -109,7 +109,8 @@ fn is_image_opaque(format: webrender_api::ImageFormat, bytes: &[u8]) -> bool {
 fn premultiply(data: &mut [u8]) {
     let length = data.len();
 
-    for i in Iterator::step_by(0..length, 4) {
+    let mut i = 0;
+    while i < length {
```
Comment (nox) [inline components/net/image_cache.rs]: What about using [`<[_]>::chunks`](https://doc.rust-lang.org/std/primitive.slice.html#method.chunks)?

File: components/net_traits/image/base.rs [hunk 2]
```diff
@@ -50,14 +50,17 @@ pub struct ImageMetadata {
 fn byte_swap_and_premultiply(data: &mut [u8]) {
     let length = data.len();
 
-    for i in Iterator::step_by(0..length, 4) {
+    let mut i = 0;
+    while i < length {
```
Comment (nox) [inline components/net_traits/image/base.rs]: What about using [`<[_]>::chunks`](https://doc.rust-lang.org/std/primitive.slice.html#method.chunks)?

File: components/script/dom/element.rs [hunk 3]
```diff
@@ -213,10 +212,8 @@ pub enum AdjacentPosition {
     BeforeEnd,
 }
 
-impl<'a> TryFrom<&'a str> for AdjacentPosition {
-    type Error = Error;
-
-    fn try_from(position: &'a str) -> Result<AdjacentPosition, Self::Error> {
+impl AdjacentPosition {
+    fn try_from(position: &str) -> Result<AdjacentPosition, Error> {
```
Comment (nox) [inline components/script/dom/element.rs]: Could you use [`FromStr`](https://doc.rust-lang.org/std/str/trait.FromStr.html) here?

**Review by nox — APPROVED** [review]
3 remarks, but r=me if you don't want to make the suggested changes.

File: components/net/image_cache.rs (hunk 1 above)
Comment (SimonSapin) [reply components/net/image_cache.rs]: The last item of `chunks` can be a smaller slice, whereas this loop will stop before.

**Pull request merged**